tests/integration/synapseclient/models/synchronous/test_permissions.py

```diff
@@ -0,0 +1,1298 @@
+            "UPDATE",
+            "CHANGE_PERMISSIONS",
+            "CREATE",
+            "MODERATE",
+            "DOWNLOAD",
+        ]
+        assert set(expected_permissions) == set(permissions)
+
+    def test_get_acl_read_only_permissions_on_entity(self) -> None:
+        # GIVEN a project created with default permissions of administrator
+        project_with_read_only_permissions = Project(
+            name=str(uuid.uuid4()) + "test_get_acl_read_permissions_on_project"
+        ).store()
+        self.schedule_for_cleanup(project_with_read_only_permissions.id)
+
+        # AND the user that created the project
+        p1 = UserProfile().get()
+
+        # AND the permissions for the user on the entity are set to READ only
+        project_with_read_only_permissions.set_permissions(
+            principal_id=p1.id, access_type=["READ"]
+        )
+
+        # WHEN I get the permissions for the user on the entity
+        permissions = project_with_read_only_permissions.get_acl(principal_id=p1.id)
+
+        # THEN I expect to see read only permissions
+        expected_permissions = ["READ"]
+        assert set(expected_permissions) == set(permissions)
+
+    def test_get_acl_through_team_assigned_to_user(self) -> None:
+        # GIVEN a project created with default permissions of administrator
+        project_with_permissions_through_single_team = Project(
+            name=str(uuid.uuid4())
+            + "test_get_acl_through_team_assigned_to_user_and_project"
+        ).store()
+
+        # AND the user that created the project
+        p1 = UserProfile().get()
+
+        name = TEAM_PREFIX + str(uuid.uuid4())
+        team = Team(
+            name=name,
+            description=DESCRIPTION_FAKE_TEAM_SINGLE_PROJECT,
+        ).create()
+
+        # Handle Cleanup - Note: When running this schedule for cleanup order
+        # can matter when there are dependent resources
+        self.schedule_for_cleanup(team.id)
+        self.schedule_for_cleanup(project_with_permissions_through_single_team.id)
+
+        # AND the permissions for the Team on the entity are set to all permissions except for DOWNLOAD
+        project_with_permissions_through_single_team.set_permissions(
+            principal_id=team.id,
+            access_type=[
+                "READ",
+                "DELETE",
+                "CHANGE_SETTINGS",
+                "UPDATE",
+                "CHANGE_PERMISSIONS",
+                "CREATE",
+                "MODERATE",
+            ],
+        )
+
+        # AND the permissions for the user on the entity are set to NONE
+        project_with_permissions_through_single_team.set_permissions(
+            principal_id=p1.id, access_type=[]
+        )
+
+        # WHEN I get the permissions for the user on the entity
+        permissions = project_with_permissions_through_single_team.get_acl(
+            principal_id=p1.id
+        )
+
+        # THEN I expect to see the permissions of the team
```

Comment:
If it's not too much trouble (and maybe this can be for another PR), I'd like to see some tests where permissions are revoked to confirm the ACL gets updated accordingly. So for this example, can we confirm that re-setting permissions such that READ and DOWNLOAD are removed from the access types reflects on the ACL?